### PR TITLE
feat: Add images with active storage and create all shack views

### PR DESCRIPTION
**Pull request by jesusMava**

### CODE REVIEW

File: app/controllers/shacks_controller.rb [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+  load_and_authorize_resource
+  protect_from_forgery prepend: true
 
-  def index; end
+  def index
+     @shack = Shack.all
+  end
+
+  def show
+    @shack = Shack.find(params[:id])
```
Comment (YourSouLi5Mine) [inline app/controllers/shacks_controller.rb:10]: I would recommend to extract the repeated `Shack.find(params[:id])` in a private method and adding a `before_action only` for the methods that will need it like show, edit, etc

File: app/models/shack.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
 class Shack < ApplicationRecord
   belongs_to :user
+
+  has_many_attached :images
+
+  validates :user, presence: true
+  validates :room_numbers, numericality: { only_integer: true, greater_than: 0 }, presence: true
+  validates :sale_price, numericality: { only_integer: true, greater_than: 0 }, presence: true
+  validates :rent_price, numericality: { only_integer: true, greater_than: 0 }, presence: true
```
Comment (YourSouLi5Mine) [inline app/models/shack.rb:9]: Is it possible to just rent my shack or sale (not both?) -> Business logic question

File: config/routes.rb [hunk 3]
```diff
@@ -4,11 +4,11 @@
   }
 
   root 'home#index'
-  resources :shacks
 
   devise_scope :user do  
     get '/users/sign_out' => 'devise/sessions#destroy'     
     get '/users' => 'devise/sessions#new'     
+    resources :shacks
```
Comment (YourSouLi5Mine) [inline config/routes.rb:11]: Great!

File: spec/models/shack_spec.rb [hunk 4]
```diff
@@ -1,5 +1,56 @@
 require 'rails_helper'
 
 RSpec.describe Shack, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it { should validate_presence_of(:description) }
```
Comment (YourSouLi5Mine) [inline spec/models/shack_spec.rb:4]: Perfect validations

File: app/models/shack.rb (hunk 2 above)
Comment (jesusMava) [reply app/models/shack.rb:9]: you  are right, both of them are possibles

Comment (YourSouLi5Mine) [reply app/models/shack.rb:9]: Then why would you need the presence of both fields?

File: app/controllers/shacks_controller.rb [hunk 5]
```diff
@@ -1,5 +1,56 @@
...
+  end
+
+  private
+    def shack_params
+      params.require(:shack).permit(:room_numbers,
+                                     :ubication, :sale_price,
+                                     :rent_price, :description,
+                                     :status, :discount,
+                                     images: []
+                                   )
```
Comment (YourSouLi5Mine) [inline app/controllers/shacks_controller.rb:50]: Check Eloquent, this looks bad

File: spec/factories/shacks.rb [hunk 6]
```diff
@@ -2,9 +2,22 @@
...
+
+    factory :images_shack, class: 'Shack' do
+      after(:build) do |shack|
+        shack.images.attach(io: File.open(Rails.root.join(
+                                            'spec', 'factories',
+                                            'images', 'Linux.png'
+                                          )),
+                                          filename: 'Linux.png',
+                                          content_type: 'image/png'
+                           )
```
Comment (YourSouLi5Mine) [inline spec/factories/shacks.rb:19]: This too looks bad

File: spec/models/shack_spec.rb [hunk 7]
```diff
@@ -1,5 +1,68 @@
...
+  end
+
+  context "images validation" do
+    let!(:user) { create(:seller) }
+    let!(:shack) { create(:shack) }
+
+    subject { create(:images_shack).images }
+
+    it "relation images" do
+      is_expected.to be_an_instance_of(ActiveStorage::Attached::Many)
```
Comment (YourSouLi5Mine) [inline spec/models/shack_spec.rb:65]: Well done!

**Pull request merged**